prow/cmd/checkconfig/main.go

```diff
@@ -793,6 +847,84 @@ func validateNeedsOkToTestLabel(cfg *config.Config) error {
 	return errorutil.NewAggregate(queryErrors...)
 }
 
+type filePresenceChecker interface {
+	remoteCheck(path string) (bool, error)
+	localCheck(path string, client *git.Client) bool
+}
+
+type _filePresenceChecker string
+
+func (f _filePresenceChecker) remoteCheck(path string) (bool, error) {
+	blobUrl := fmt.Sprintf("https://github.com/%s/blob/master/%s", f, path)
+	resp, err := http.Head(blobUrl)
+	if err != nil {
+		return false, err
+	}
+	return resp.StatusCode == 200, nil
+}
+
+func (f _filePresenceChecker) localCheck(path string, gc *git.Client) bool {
+	r, err := gc.Clone(string(f))
+	if err != nil {
+		return false
+	}
+	_, err = os.Stat(filepath.Join(r.Dir, path))
+	return err == nil
+}
+
+func hasOwnersFile(repo filePresenceChecker, gc *git.Client) (bool, error) {
+	exists, err := repo.remoteCheck("OWNERS")
+	if err != nil {
+		return false, err
+	}
+	if exists {
+		return true, nil
+	}
+
+	return repo.localCheck("OWNERS", gc), nil
+}
+
+func verifyOwnersPresence(cfg *plugins.Configuration, rc github.RepositoryClient, gc *git.Client) error {
+	approveConfig := enabledOrgReposForPlugin(cfg, approve.PluginName, false)
```

Comment:
Shouldn't we do similar logic to the verify-owners check here to get the full set of repos?

Reply:
I was following the definition in the card which was only talking about `approve` plugin, but it should not be hard to enhance the check for other plugins.

Reply:
I did what you proposed, and the interesting result for OpenShift config is that we end up cloning and checking almost everything from `redhat-developer` and `openshift` org, because we enable `blunderbuss` and `owners-label` plugins org-wide, but many repos (old, one-off, etc) do not have `OWNERS` present.